src/window/gui/Gui.h

```diff
@@ -72,48 +72,56 @@ class Gui {
     ~Gui();
 
     void Init(GuiWindowInitData windowImpl);
-    void StartFrame();
-    void EndFrame();
-    void RenderViewports();
-    void DrawMenu();
+    void Draw();
+    void HandleWindowEvents(WindowEvent event);
+    void SaveConsoleVariablesNextFrame();
+    bool SupportsViewports();
 
-    void SaveConsoleVariablesOnNextTick();
-    void Update(WindowEvent event);
     void AddGuiWindow(std::shared_ptr<GuiWindow> guiWindow);
+    std::shared_ptr<GuiWindow> GetGuiWindow(const std::string& name);
     void RemoveGuiWindow(std::shared_ptr<GuiWindow> guiWindow);
     void RemoveGuiWindow(const std::string& name);
     void RemoveAllGuiWindows();
+
     void LoadGuiTexture(const std::string& name, const std::string& path, const ImVec4& tint);
     bool HasTextureByName(const std::string& name);
     void LoadGuiTexture(const std::string& name, const LUS::Texture& tex, const ImVec4& tint);
     void UnloadTexture(const std::string& name);
     ImTextureID GetTextureByName(const std::string& name);
     ImVec2 GetTextureSize(const std::string& name);
-    bool SupportsViewports();
-    std::shared_ptr<GuiWindow> GetGuiWindow(const std::string& name);
+    void LoadTextureFromRawImage(const std::string& name, const std::string& path);
+
     std::shared_ptr<GameOverlay> GetGameOverlay();
     void SetMenuBar(std::shared_ptr<GuiMenuBar> menuBar);
     std::shared_ptr<GuiMenuBar> GetMenuBar();
-    void LoadTextureFromRawImage(const std::string& name, const std::string& path);
-    bool ImGuiGamepadNavigationEnabled();
-    void BlockImGuiGamepadNavigation();
-    void UnblockImGuiGamepadNavigation();
     void SetMenu(std::shared_ptr<GuiWindow> menu);
-    bool GetMenuOrMenubarVisible();
     std::shared_ptr<GuiWindow> GetMenu();
-    bool GetPadBtnTogglesMenu();
-    void SetPadBtnTogglesMenu();
+    bool GetMenuOrMenubarVisible();
 
-    void ImGuiBackendNewFrame();
-    void ImGuiWMNewFrame();
+    bool GamepadNavigationEnabled();
+    void BlockGamepadNavigation();
+    void UnblockGamepadNavigation();
 
   protected:
+    void StartFrame();
+    void EndFrame();
+    void DrawFloatingWindows();
+    void DrawMenu();
+    void DrawGame();
+
+    void ImGuiBackendNewFrame();
+    void ImGuiWMNewFrame();
     void ImGuiWMInit();
     void ImGuiBackendInit();
     void ImGuiRenderDrawData(ImDrawData* data);
+
+    bool GetPadBtnTogglesMenu();
+    void SetPadBtnTogglesMenu(bool padBtnTogglesMenu);
```

Comment:
You moved `SetPadBtnTogglesMenu` to `protected`, but 2ship calls it externally for the Modern Menu. /cc @Malkierian

Reply:
For these to be public and maintained long term, I'd rather we set the bad button via these functions instead of the boolean.

Reply:
This isn't for setting what pad button opens the menu, this is for switching between menubar and menu. If we end up completely yeeting the menubar in the future, then this becomes unnecessary. But for now, the choice is still up to the ports, thus the option needs to be toggleable.